src/Containers/Impl/IContainer.cs

```diff
@@ -0,0 +1,9 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.R.Containers {
+    public interface IContainer {
+        string Id { get; set; }
```

Comment:
May be worth adding IsMatch12 and IsMatch64 and have service methods return IContainer and hide id matching impementation.